providers/netcore/src/lib/netcore.provider.ts

```diff
@@ -22,76 +24,84 @@ export enum NetCoreStatusEnum {
     }
-  ) {}
+  ) {
+    this.axiosInstance = axios.create({
+      baseURL: this.BASE_URL,
+    });
+  }
 
   async sendMessage(
     options: IEmailOptions
   ): Promise<ISendMessageSuccessResponse> {
-    const netcoreLib = await import('pepipost/lib');
-
-    netcoreLib.Configuration.apiKey = this.config.apiKey;
-
-    const controller = netcoreLib.MailSendController;
-    const body = new netcoreLib.Send();
-
-    body.from = new netcoreLib.From();
-    body.from.email = options.from || this.config.from;
-    body.subject = options.subject;
-
-    body.content = [];
-    body.content[0] = new netcoreLib.Content();
-    body.content[0].type = netcoreLib.TypeEnum.HTML;
-    body.content[0].value = options.html;
-
-    body.personalizations = [];
-    body.personalizations[0] = new netcoreLib.Personalizations();
-    body.personalizations[0].to = options.to.map((email) => {
-      const item = new netcoreLib.EmailStruct();
-      item.email = email;
+    const data: INetCoreEmail['body'] = {
+      from: { email: options.from || this.config.from },
+      subject: options.subject,
+      content: [
+        {
+          type: 'html',
+          value: options.html,
+        },
+      ],
+      personalizations: [
+        {
+          to: options.to.map((email) => ({ email })),
+        },
+      ],
+    };
 
-      return item;
-    });
+    if (options.replyTo) {
+      data.reply_to = options.replyTo;
+    }
 
     if (options.cc) {
-      body.personalizations[0].cc = options.cc.map((ccItem, index) => {
-        const email = new netcoreLib.EmailStruct();
-        email.email = ccItem;
-
-        return email;
-      });
+      data.personalizations[0].cc = options.cc.map((email) => ({
+        email,
+      }));
     }
 
     if (options.bcc) {
-      body.personalizations[0].bcc = options.bcc.map((ccItem, index) => {
-        const email = new netcoreLib.EmailStruct();
-        email.email = ccItem;
+      data.personalizations[0].bcc = options.bcc.map((email) => ({
+        email,
+      }));
+    }
 
-        return email;
-      });
+    if (options.attachments) {
+      data.personalizations[0].attachments = options.attachments?.map(
+        (attachment) => {
+          return {
+            name: attachment.name,
+            content: attachment.file.toString('base64'),
+          };
+        }
+      );
     }
 
-    body.personalizations[0].attachments = options.attachments?.map(
-      (attachment) => {
-        const attachmentPayload = new netcoreLib.Attachments();
-        attachmentPayload.content = attachment.file.toString('base64');
-        attachmentPayload.filename = attachment.name;
+    const emailOptions = {
+      method: 'POST',
+      headers: {
+        api_key: this.config.apiKey,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      data,
+    };
 
-        return attachment;
-      }
+    const response: INetCoreEmail['response'] = await this.axiosInstance.post(
```

Comment:
I wonder i we could remove those square brackets with dynamic string and switch it to dot notation, same for all of the `INetCoreEmail[]` use.